internal/storage/serde.go

```diff
@@ -237,9 +237,6 @@ func (deser *DeserializeReader[T]) Next() error {
 			return err
 		}
 		deser.pos = 0
-		if deser.rec != nil {
-			deser.rec.Release()
```

Comment:
`deser.rec` should not be released here, because it is reused in the entire read process. 

This issue is found in benchmark.